From 3e6013d397983cfb662f8eeeb000d5587e345304 Mon Sep 17 00:00:00 2001
From: yoinx <joseph.schubert@gmail.com>
Date: Mon, 13 Feb 2017 02:24:45 -0700
Subject: [PATCH 2/3] HBM: Fix persistence across suspend

Certain devices like the Nexus 6P will reset this sys value on suspend
so we'll add this hook to copy the value again if it was enabled before
the display was turned off

Change-Id: Ia8d7af8205ea029f187676d98e59306e20a596fa
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 services/core/java/com/android/server/power/PowerManagerService.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/services/core/java/com/android/server/power/PowerManagerService.java b/services/core/java/com/android/server/power/PowerManagerService.java
index 897dc3ca8748..e88915e11994 100644
--- a/services/core/java/com/android/server/power/PowerManagerService.java
+++ b/services/core/java/com/android/server/power/PowerManagerService.java
@@ -2318,6 +2318,10 @@ public final class PowerManagerService extends SystemService
             screenAutoBrightnessAdjustment = Math.max(Math.min(
                     screenAutoBrightnessAdjustment, 1.0f), -1.0f);
 
+            if (mSupportsHighBrightnessModeConfig && mHighBrightnessModeEnabled) {
+                nativeSetFeature(POWER_FEATURE_HIGH_BRIGHTNESS_MODE, 1);
+            }
+
             // Update display power request.
             mDisplayPowerRequest.screenBrightness = screenBrightness;
             mDisplayPowerRequest.screenAutoBrightnessAdjustment =
-- 
2.13.0

